tests/integration/test_recipes_loading.py

```diff
@@ -1,157 +1,77 @@
-@pytest.fixture
-def patched_extract_shape(monkeypatch):
-    """Replace `_recipe_checks.extract_shape`.
-
-    Skips check that shapefile exists.
-    """
-    def extract_shape(settings):
-        valid = {
-            'method': {'contains', 'representative'},
-            'crop': {True, False},
-        }
-
-        for key in valid:
-            value = settings.get(key)
-            if not (value is None or value in valid[key]):
-                raise _recipe_checks.RecipeError(
-                    "In preprocessor function `extract_shape`: Invalid value"
-                    f"'{value}' for argument '{key}', choose from "
-                    "{}".format(', '.join(f"'{k}'".lower()
-                                          for k in valid[key])))
-
-    monkeypatch.setattr(_recipe_checks, 'extract_shape', extract_shape)
-
-
-@pytest.fixture
-def patched_get_reference_levels(monkeypatch):
-    """Replace `_regrid.get_reference_levels`.
+def _get_recipes():
+    recipes_path = Path(esmvaltool.__file__).absolute().parent / 'recipes'
+    recipes = tuple(recipes_path.glob("**/recipe*.yml"))
+    ids = tuple(str(p.relative_to(recipes_path)) for p in recipes)
+    return recipes, ids
 
-    Return a random set of reference levels
-    """
-    def get_reference_levels(*_, **_a):
-        return [1, 2]
 
-    monkeypatch.setattr(esmvalcore._recipe, 'get_reference_levels',
-                        get_reference_levels)
+RECIPES, IDS = _get_recipes()
 
 
-@pytest.mark.parametrize('recipe_file', _get_recipes())
-def test_diagnostic_run(recipe_file, config_user, patched_datafinder,
-                        patched_extract_shape, patched_get_reference_levels):
+@pytest.mark.parametrize('recipe_file', RECIPES, ids=IDS)
+def test_recipe_valid(recipe_file, config_user, monkeypatch):
     """Check that recipe files are valid ESMValTool recipes."""
-    read_recipe_file(recipe_file, config_user)
+    # Mock input files
+    find_files = create_autospec(esmvalcore._data_finder.find_files,
+                                 spec_set=True)
+    find_files.side_effect = lambda *_, **__: [
+        'test_0000-1849.nc',
+        'test_1850-9999.nc',
+    ]
+    monkeypatch.setattr(esmvalcore._data_finder, 'find_files', find_files)
+
+    # Mock vertical levels
+    levels = create_autospec(esmvalcore._recipe.get_reference_levels,
+                             spec_set=True)
+    levels.side_effect = lambda *_, **__: [1, 2]
+    monkeypatch.setattr(esmvalcore._recipe, 'get_reference_levels', levels)
+
+    # Mock valid NCL version
+    ncl_version = create_autospec(esmvalcore._recipe_checks.ncl_version,
+                                  spec_set=True)
+    monkeypatch.setattr(esmvalcore._recipe_checks, 'ncl_version', ncl_version)
+
+    # Mock interpreters installed
+    def which(executable):
+        if executable in ('julia', 'ncl', 'python', 'Rscript'):
+            path = '/path/to/' + executable
+        else:
+            path = None
+        return path
```

Comment:
have written exactly the same bit (even `/path/to/ + ...`) for the test_task in core a few days ago :grin: